packages/playwright-core/src/server/injected/selectorGenerator.ts

```diff
@@ -216,8 +216,12 @@ function buildCandidates(injectedScript: InjectedScript, element: Element, testI
     candidates.push({ engine: 'css', selector: `${cssEscape(element.nodeName.toLowerCase())}[name=${quoteAttributeValue(element.getAttribute('name')!)}]`, score: kCSSInputTypeNameScore });
 
   if (element.getAttribute('title')) {
-    candidates.push({ engine: 'internal:attr', selector: `[title=${escapeForAttributeSelector(element.getAttribute('title')!, false)}]`, score: kTitleScore });
-    candidates.push({ engine: 'internal:attr', selector: `[title=${escapeForAttributeSelector(element.getAttribute('title')!, true)}]`, score: kTitleScoreExact });
+    if (element.nodeName === 'IFRAME') {
```

Comment:
Do we need a nodeName === IFRAME check as a first thing here given that we won't be using any of the non-CSS engines to locate it?